tensorflow_datasets/structured/heart_disease.py

```diff
@@ -0,0 +1,132 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Heart disease dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """\
+@misc{Dua:2019 ,
+author = "Janosi, Steinbrunn and Pfisterer, Detrano",
+year = "1988",
+title = "{UCI} Machine Learning Repository",
+url = "http://archive.ics.uci.edu/ml/datasets/Heart+Disease",
+institution = "University of California, Irvine, School of Information and Computer Sciences"
+}
+"""
+
+_DESCRIPTION = """\
+This data set contain 13 attributes and labels of heart disease from \
+303 participants from Cleveland since Cleveland data was most commonly\
+used in modern research.
+
+Attribute by column index
+1. age      : age in years
+2. sex      : sex (1 = male; 0 = female)
+3. cp       : chest pain type
+    (1 = typical angina; 2 = atypical angina; 3 = non-anginal pain; 4 = asymptomatic)
+4. trestbps : resting blood pressure (in mm Hg on admission to the hospital)
+5. chol     : serum cholestoral in mg/dl
+6. fbs      : (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
+7. restecg  : resting electrocardiographic results
+8. thalach  : maximum heart rate achieved
+9. exang    : exercise induced angina (1 = yes; 0 = no)
+10. oldpeak : ST depression induced by exercise relative to rest
+11. slope   : the slope of the peak exercise ST segment (1 = upsloping; 2 = flat; 3 = downsloping)
+12. ca      : number of major vessels (0-3) colored by flourosopy
+13. thal    : 3 = normal; 6 = fixed defect; 7 = reversable defect
+14. num (the predicted attribute): diagnosis of heart disease (angiographic disease status)
+    (0 = < 50% diameter narrowing, no presence of heart disease;
+     1 = > 50% diameter narrowing, with increasing severity)
+Dataset Homepage: http://archive.ics.uci.edu/ml/datasets/Heart+Disease
+"""
+
+_DOWNLOAD_URL = 'http://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/processed.cleveland.data'
+
+
+class HeartDisease(tfds.core.GeneratorBasedBuilder):
+  """Heart disease dataset with 13 attributes."""
+
+  VERSION = tfds.core.Version("0.0.1", "New split API (https://tensorflow.org/datasets/splits)")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "features":
+                collections.OrderedDict([
+                    ("age", tf.int32),
+                    ("sex", tf.int32),
```

Comment:
For categorical features, is it possible to use `ClassLabel` instead ? So users can get the semantic label ? `inffo.features['feature']['sex'].names == ['female', 'male']`

Reply:
Ok,  I'll change it.

Reply:
Added `ClassLabel` wherever possible. Done